charts/argo-workflows/templates/post-sync-workflow.yaml

```diff
@@ -5,6 +5,9 @@ metadata:
 spec:
   entrypoint: handle-sync-event
   onExit: exit-handler
+  retryStrategy:
+    limit: "3"
+    expression: "asInt(lastRetry.exitCode) > 1"
```

Comment:
The `> 1` is a little surprising. Does the job return 2 or greater when it fails?

Reply:
Yes it's surprising, not sure why but it used exit code 2:

```
STEP                       TEMPLATE                   PODNAME                     DURATION  MESSAGE
 ✖ post-sync-nb66d         handle-sync-event                                                                     
 └─✖ smoke-test            smoke-test/smoke-test      post-sync-nb66d-402463690   38m       Error (exit code 2)  
```

Reply:
Output of `kubectl -napps describe pod post-sync-nb66d-402463690` below.

```
    State:          Terminated
      Reason:       Error
      Exit Code:    2
```